### DISCUSSION TITLE
Installation error on Expanse cluster at SDSC

### DISCUSSION THREAD

**Discussion by tengzhang48** [idaholab/moose#18864] [Q&A Getting Started] [answered]
Hello,

I followed the online instruction to install Moose on Expanse cluster at SDSC. When compiling libmesh, I got the following error
"  CXX      apps/version.o
  CXX      utilities/src/metaphysicl_version.lo
  CXXLD    libmetaphysicl.la
/usr/bin/ld: cannot find -lnuma
collect2: error: ld returned 1 exit status
make[4]: *** [Makefile:710: libmetaphysicl.la] Error "

This is related to the function of "metaphysical", which is very interesting to me.

Did I miss some modules to load on the cluster? So far, I have just followed the online instruction. I would like to hear suggestions from the Moose community and then ask for more specific help from the Expanse support if needed.

Thanks.

Best,
Teng

**Comment by roystgnr** [comment, marked as answer]
libMesh and MetaPhysicL have nothing directly to do with libnuma, and to the best of my knowledge neither does Moose ... but we do see these sorts of errors all the time, when an MPI or a PETSc installation is for some reason tied to libraries that don't exist, then the MPI compiler wrapper tries to link them (or we parse PETSc configure output and find linker arguments that try to link them) and the link command fails.

We usually see those errors at configure time, though, either the first time the configure script tries to build an executable with mpicxx or the first time it tries to build a PETSc-linked test executable.  I'm not sure how you could have made it through both libMesh and MetaPhysicL configure scripts without failure only to see a failure here.  You might need to post your config logs before we can say what's going on.

**Reply by tengzhang48** [reply]
Thanks for your information. The Expanse support team helped me figure the problem. After explicitly exporting the path of "numactl", I can compile the software.

Best,
Teng
